src/main/webapp/courseLoader.js

```diff
@@ -271,11 +274,8 @@
   }
 
   function newURL(
```

Comment:
it doesn't seem like you use the newURL function anywhere other than right above this, so why don't you only set one parameter for the function (whatever ratingProperties is) and then call each specific attribute inside the function itself